src/main/java/org/ecocean/servlet/ServletUtilities.java

```diff
@@ -44,9 +44,113 @@
 import javax.servlet.http.Cookie;
 
 import org.apache.commons.lang3.StringEscapeUtils;
+import org.apache.commons.text.WordUtils;
+import org.ecocean.CommonConfiguration;
+import org.ecocean.ContextConfiguration;
+import org.ecocean.Organization;
+import org.ecocean.servlet.ServletUtilities;
+import org.ecocean.Shepherd;
+import org.ecocean.ShepherdProperties;
+import org.ecocean.User;
+import org.ecocean.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ServletUtilities {
     public static String getHeader(HttpServletRequest request) {
+        String context = "context0";
+
+        context = ServletUtilities.getContext(request);
+        String langCode = ServletUtilities.getLanguageCode(request);
+        Properties props = new Properties();
+        props = ShepherdProperties.getProperties("header.properties", langCode, context);
+        // System.out.println("props" + props);
+        String urlLoc = "//" + CommonConfiguration.getURLLocation(request);
+        String pageTitle = (String)request.getAttribute("pageTitle"); // allows custom override from calling jsp (must
+        String siteName = "wildbook";
+        try {
+            // System.out.println("getting site name ");
+            String name = props.getProperty("siteName");
+            // System.out.println("site name is " + name);
+            if (name != null) {
+                siteName = name;
+            }
+        } catch (Exception e) {
+            System.out.println("Failed to get siteName property from props:");
+            e.printStackTrace();
+        }
+        // set BEFORE include:header)
+        if (pageTitle == null) {
+            pageTitle = CommonConfiguration.getHTMLTitle(context);
+        } else {
+            pageTitle = CommonConfiguration.getHTMLTitle(context) + " | " + pageTitle;
+        }
+        String username = null;
+        User user = null;
+        String profilePhotoURL = urlLoc + "/images/empty_profile.jpg";
+        // we use this arg bc we can only log out *after* including the header on
+        // logout.jsp. this way we can still show the logged-out view in the header
+        boolean loggingOut = Util.requestHasVal(request, "loggedOut");
+        String notifications = "";
+        // check if user is logged in and has pending notifications
+        if (request.getUserPrincipal() != null) {
+            Shepherd myShepherd = new Shepherd(context);
+            myShepherd.setAction("header.jsp");
+            myShepherd.beginDBTransaction();
+            try {
+                notifications = Collaboration.getNotificationsWidgetHtml(request, myShepherd);
+                if (request.getUserPrincipal() != null && !loggingOut) {
+                    user = myShepherd.getUser(request);
+                    username = (user != null) ? user.getUsername() : null;
```

Comment:
might be best to wrap this whole section in `if (user != null) { ....` rather than test for `username` separately, since we also should have one for the line below: `if (user.getUserImage()...)` is _not_ checking for null `user`.